packages/rest/test/unit/coercion/paramStringToNumber.unit.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {test} from './utils';
+import {ParameterLocation} from '@loopback/openapi-v3-types';
+import * as HttpErrors from 'http-errors';
+import {HttpErrorMessage} from './../../../';
+
+const NUMBER_PARAM = {
+  in: <ParameterLocation>'path',
+  name: 'aparameter',
+  schema: {type: 'number'},
+};
+
+const REQUIRED_NUMBER_PARAM = {
+  in: <ParameterLocation>'path',
+  name: 'aparameter',
+  schema: {type: 'number'},
+  required: true,
+};
+
+const FLOAT_PARAM = {
+  in: <ParameterLocation>'path',
+  name: 'aparameter',
+  schema: {
+    type: 'number',
+    float: 'float',
+  },
+};
+
+const DOUBLE_PARAM = {
+  in: <ParameterLocation>'path',
+  name: 'aparameter',
+  schema: {
+    type: 'number',
+    float: 'double',
+  },
+};
+
+describe('coerce param from string to number - required', () => {
+  context('valid values', () => {
+    test(REQUIRED_NUMBER_PARAM, '0', 0);
+    test(REQUIRED_NUMBER_PARAM, '1', 1);
+    test(REQUIRED_NUMBER_PARAM, '-1', -1);
+  });
+
+  context('empty values trigger ERROR_BAD_REQUEST', () => {
+    // null, '' sent from request are converted to raw value ''
+    test<HttpErrors.HttpError>(
```

Comment:
Is this explicit type required? Cannot TypeScript infer the type for us?

Reply:
oops, forgot to remove it.